cpp/CMakeLists.txt

```diff
@@ -37,12 +36,15 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     "Debug" "Release")
 endif()
 
-###################################################################################################
-# - User Options  ---------------------------------------------------------------------------------
+##############################################################################
+# - User Options  ------------------------------------------------------------
 
-option(BUILD_CUML_CPP_LIBRARY "Build libcuml++ shared library" ON)
+set(BLAS_LIBRARIES "" CACHE STRING
+    "Location of BLAS library for FAISS build.")
 
-option(BUILD_CUML_C_LIBRARY "Build libcuml shared library" ON)
+option(BUILD_CUML_C_LIBRARY "Build libcuml shared library. Contins the cuML C API" ON)
```

Comment:
The C and CPP libs seems more like make targets than cmake options, but I'm not as familiar with the history.

Reply:
It's just so that you can specify at cmake time that you don't want to build something in that build folder at all, so that `make -j` doesn't compile things you don't want later on. Mainly I added those for me since I tend to have a lot of build folders, and by specifying this at cmake time, I don't have to remember which target I need to build in which folder